lib/import/brca/providers/bristol/bristol_handler.rb

```diff
@@ -6,43 +6,67 @@ module Providers
-            process_cdna_change(genotype, record)
-            add_protein_impact(genotype, record)
-            add_organisationcode_testresult(genotype)
-            @persister.integrate_and_store(genotype)
-          end
-
-          def add_organisationcode_testresult(genotype)
-            genotype.attribute_map['organisationcode_testresult'] = '698V0'
-          end
-
-          def add_simple_fields(genotype, record)
             genotype.add_passthrough_fields(record.mapped_fields,
                                             record.raw_fields,
                                             PASS_THROUGH_FIELDS)
-            # extractGeneAndLocation(record.raw_fields['chrpos'], genotype)
-            genotype.add_gene(record.mapped_fields['gene'].to_i)
+            process_cdna_change(genotype, record)
+            add_protein_impact(genotype, record)
+            add_organisationcode_testresult(genotype)
             genotype.add_gene_location(record.mapped_fields['codingdnasequencechange'])
-            genotype.add_protein_impact(record.mapped_fields['proteinimpact'])
+            add_protein_impact(genotype, record)
             genotype.add_variant_class(record.mapped_fields['variantpathclass'])
             genotype.add_received_date(record.raw_fields['received date'])
             process_genomic_change(genotype, record)
             genotype.add_test_scope(:full_screen)
+            process_test_status(genotype, record)
             genotype.add_method('ngs')
+            res = process_gene(genotype, record)
+            res&.each { |cur_genotype| @persister.integrate_and_store(cur_genotype) }
           end
+          # rubocop:enable  Metrics/MethodLength
+          # rubocop:enable  Metrics/AbcSize
+
+          def add_organisationcode_testresult(genotype)
+            genotype.attribute_map['organisationcode_testresult'] = '698V0'
+          end
+
+          def process_negative_record(record, negative_gene, genotypes)
+            duplicated_genotype = Import::Brca::Core::GenotypeBrca.new(record)
+            duplicated_genotype.add_gene(negative_gene.join)
+            duplicated_genotype.add_status(1)
+            duplicated_genotype.add_test_scope(:full_screen)
+            duplicated_genotype.add_passthrough_fields(record.mapped_fields,
+                                                       record.raw_fields,
+                                                       PASS_THROUGH_FIELDS)
+            genotypes.append(duplicated_genotype)
+          end
+
+          def process_gene(genotype, record)
+            return if record.raw_fields['gene'].nil?
+
+            genotypes = []
+            negative_gene = %w[BRCA1 BRCA2] - [record.raw_fields['gene']]
+            process_negative_record(record, negative_gene, genotypes)
+            genotype.add_gene(record.raw_fields['gene'])
+            genotypes.append(genotype)
+          end
+
+          # rubocop:disable Style/GuardClause
```

Comment:
@FrancescoNea I'm not sure that this cop needs to be disabled, it looks correctable?